### PR TITLE
Easier calendar navigation

### LINKED ISSUE #277: Easier calendar navigation
Some things could be done to make calendar navigation easier. These come to mind:

- [x] Week and month calendar: Highlight _today_ with different color
- [x] Week and month calendar: Add button to navigate to _today_
- [x] Week calendar: Differentiate night from day (7.00-22.00?) using the background color
- [x] Week calendar: Differentiate past days from future (e.g. fade out weekday/date labels slightly)

### PR DESCRIPTION
**Pull request by omidhq**
Resolves #277

### CODE REVIEW

File: src/components/ZetkinCalendar/WeekCalendar/index.tsx
Comment (richardolsson) [inline src/components/ZetkinCalendar/WeekCalendar/index.tsx:121]: The complicated date constructor here is identical to `currentDate`, which could be used here instead.

File: src/components/ZetkinCalendar/WeekCalendar/index.tsx
Comment (richardolsson) [inline src/components/ZetkinCalendar/WeekCalendar/index.tsx:128]: Same is true here I believe, use `currentDate` instead.

File: src/components/ZetkinCalendar/index.tsx
Comment (richardolsson) [inline src/components/ZetkinCalendar/index.tsx:86]: I feel like this is unnecessarily complex, and also doesn't behave the way I would expect.

If today's is last of July, it will be visible in the month view UI even if the currently focused month is August (because August starts on a Sunday and displays the full preceding week of July as well). But this function will return `false`. I would not have considered that desired behavior. But that's my personal opinion and it could very well be argued that it is exactly how it should work.

A less complex way IMO would be to check whether `firstDateInView < today < lastDateInView`. The `firstDateInView` and `lastDateInView` are already calculated in the week and month views and that logic could be moved to utility functions that are reused here.

But it's working, and the code is good, so lets keep it like this for now. No changes needed!

**Review by richardolsson — CHANGES_REQUESTED** [review]
Great work! I do have some feedback below and a number of things that I think should be changed before we merge this. It's all just small things, so hopefully something you can fix before the end of today?

File: src/components/ZetkinCalendar/WeekCalendar/index.tsx
Comment (nehahirve) [reply src/components/ZetkinCalendar/WeekCalendar/index.tsx:106]: agree, my bad for suggesting `tomorrow` as the variable name

File: src/components/ZetkinCalendar/index.tsx
Comment (nehahirve) [reply src/components/ZetkinCalendar/index.tsx:86]: Is `dayjs().week()` just getting the current week number? 
If so, I wouldn't mix it with the polyfill `.getWeekNumber()` -- as discussed in https://github.com/zetkin/app.zetkin.org/issues/329 there's a bug in this, and it will return the wrong week from 2022 onwards. 

I'd use the polyfill twice, or dayJS twice.

I also agree w @richardolsson that is todayInView() should consider the entire calendar view, not just the month start and end. Otherwise we will end up with two blue dots on the page when today is before the current month, but inside the calendar view, which would be weird.

File: src/components/ZetkinCalendar/index.tsx
Comment (nehahirve) [inline src/components/ZetkinCalendar/index.tsx:88]: Using @richardolsson 's suggestion above, you'd cut down on a lot of code by doing 
`isTodayBeforeView = today < firstCalendarDate` (i think this what the variable is called, it's the first date in the view)
`isTodayAfterView = today > lastCalendarDate`

Comment (richardolsson) [reply src/components/ZetkinCalendar/index.tsx:88]: The problem is that those variables are only available inside the distinct views, i.e. the month and week calendars respectively (and in fact the `lastCalendarDate` does not exist). So they would need to be calculated here too, e.g. using some utility function.

File: src/components/ZetkinCalendar/MonthCalendar/index.tsx
Comment (nehahirve) [inline src/components/ZetkinCalendar/MonthCalendar/index.tsx:139]: This might be overkill, maybe @richardolsson has an opinion, but, 

It's wasteful to call the isInRange function (30-odd times) when we know today isn't in the calendar view.
Maybe add logic in the top of the component that checks if today is in View and only call this function then?

smth like
`todayInView = firstCalendarDay > today > lastCalendarDay`

File: src/components/ZetkinCalendar/index.tsx
Comment (nehahirve) [reply src/components/ZetkinCalendar/index.tsx:88]: Yes, sorry, use utility functions and then do: 

`isTodayBeforeView = today < getfirstDateInView()`

File: src/components/ZetkinCalendar/index.tsx
Comment (nehahirve) [inline src/components/ZetkinCalendar/index.tsx:76]: This is a small thing (that I learnt from @richardolsson XD ) but it's worth considering when or if you use single-line if statements. 
Although it's concise and arguably as easy to read, the indents in the code provide useful visual information about the complexity of branches

so seeing 
`if (focusDate.getFullYear() > dayjs().get('year')) {
        return true
  }`
  
visually indicates that there's an execution branch there, which is nice.

File: src/components/ZetkinCalendar/WeekCalendar/index.tsx
Comment (omidhq) [reply src/components/ZetkinCalendar/WeekCalendar/index.tsx:24]: I agree with 3 and more agree with 4. Guess 6 am to 7 pm was intended as average daylight hours during the year (which of course varies)but @nehahirve might be able to explain this better as she suggested that while we pair programmed this feature.

Comment (nehahirve) [reply src/components/ZetkinCalendar/WeekCalendar/index.tsx:24]: no opinion, it was just an arbitrary decision to get things moving along, no thought went behind it!

File: src/components/ZetkinCalendar/index.tsx
Comment (richardolsson) [reply src/components/ZetkinCalendar/index.tsx:78]: Me and @omidhq solved this together in 9dc6847113d5e2215abe6aa2e2d89f3d295a2558.

File: src/components/ZetkinCalendar/index.tsx
Comment (richardolsson) [reply src/components/ZetkinCalendar/index.tsx]: See 9dc6847113d5e2215abe6aa2e2d89f3d295a2558 for simpler solution.

File: src/components/ZetkinCalendar/MonthCalendar/index.tsx
Comment (richardolsson) [reply src/components/ZetkinCalendar/MonthCalendar/index.tsx:140]: It's a good suggestion, but let's leave it as it is for now. 😊

File: src/components/ZetkinCalendar/index.tsx
Comment (richardolsson) [inline src/components/ZetkinCalendar/index.tsx:91]: This is a really long and hard to read line. It could probably be simplified greatly, but should at least be broken up on several lines.

File: src/components/ZetkinCalendar/MonthCalendar/index.tsx
Comment (richardolsson) [inline src/components/ZetkinCalendar/MonthCalendar/index.tsx:69]: Curious to hear your thoughts on this @nehahirve

**Review by richardolsson — CHANGES_REQUESTED** [review]
Just some minor simplifications needed and then I think we're ready to merge! 👇

File: src/components/ZetkinCalendar/MonthCalendar/index.tsx
Comment (nehahirve) [reply src/components/ZetkinCalendar/MonthCalendar/index.tsx:69]: it looks good, i think i'd have done similar

File: src/components/ZetkinCalendar/utils.ts
Comment (nehahirve) [inline src/components/ZetkinCalendar/utils.ts:29]: Why not use the `getNewDateWithOffset` utility function available in `dateutils.ts`? and just do `lastDayInView = getNewDateWithOffset(firstDateInView,  7)`

Comment (richardolsson) [reply src/components/ZetkinCalendar/utils.ts:29]: That's true! That would be cleaner. But I think it should be seven (not six) days because I believe the `lastDayInView` needs to be the midnight on the following Monday.

File: src/components/ZetkinCalendar/index.tsx
Comment (omidhq) [reply src/components/ZetkinCalendar/index.tsx:91]: I tried having Tooltip outside the ternary in different possible ways. It kept the tooltip visible all the time (like the way it was showing it when we were pair coding together.) Eventually, I only broke it up into multiple lines.

File: src/components/ZetkinCalendar/utils.ts
Comment (omidhq) [reply src/components/ZetkinCalendar/utils.ts:29]: I don't have `dateutils.ts` in this branch. Is it a recent merge?

**Comment by omidhq** [comment]
> Just some minor simplifications needed and then I think we're ready to merge! 👇

File: src/components/ZetkinCalendar/utils.ts
Comment (richardolsson) [reply src/components/ZetkinCalendar/utils.ts:29]: It's available on `main`. Are you comfortable merging `main` into your branch and pushing it? Otherwise lets leave it as it is for now.

**Comment by omidhq** [comment]
@richardolsson Today I got some free time to make the changes and merge this PR when I noticed this (picture👇) If you check the Week calendar on a Sunday, the InView is actually next week so it shows two blue dots when you go to the actual "today" aka Sunday view. I will convert this PR to draft for now before being able to fix this bug sometime next week. 
sc

**Comment by omidhq** [comment]
> @richardolsson Today I got some free time to make the changes and merge this PR when I noticed this (picture👇) If you check the Week calendar on a Sunday, the InView is actually next week so it shows two blue dots when you go to the actual "today" aka Sunday view. I will convert this PR to draft for now before being able to fix this bug sometime next week.
> sc

It's weird that I can't see this bug anymore. Have been hard coding the dates all morning to see it on a Sunday again but not there. Have been console.log-ing too. No luck. @richardolsson any suggestions?

**Review by richardolsson — APPROVED** [review]
Code looks good and it works as expected, except for the week number bug that is tracked by separate issue #329. It should be fixed ASAP in a separate PR.

The last two commits have some issues. The messages do not explain what they do, and unless I'm mistaken, the first of the two is not atomic and does not result in code that compiles.

Let's talk about it later. This is ready to merge!

**Pull request merged**
